Check for list of bots from an array

riyadhalnur opened the pull request:
Small update to check for bots from external services. @Tomtomgo 

**UPDATED/ADDED:**
- Check for the simple 'bot' string or from the list of bots from an array. Some bots from several services do not have the word bot in them.

**Signed-off-by:** Riyadh Al Nur riyadhalnur@verticalaxisbd.com

riyadhalnur committed:
Check for list of bots from an array

UPDATED/ADDED:
- Check for the simple 'bot' string or from the list of bots from an array. Some bots from several services do not have the word bot in them.

Signed-off-by: Riyadh Al Nur <riyadhalnur@verticalaxisbd.com>

Tomtomgo commented on index.js:
```diff
@@ -6,8 +6,10 @@
+        var bots_list = ['baiduspider', 'facebookexternalhit', 'embedly', 'quora', 'link', 'preview', 'outbrain', 'pinterest', 'vkShare', 'W3C_Validator'];
+        // Check for the simple 'bot' string or from the list of bots from an array.
+        // Some bots from several services do not have the word bot in them.
+        return user_agent.toLowerCase().indexOf('bot') > -1 || bots_list.indexOf(user_agent) > -1;
```
This won't work I think:

```
bots_list.indexOf(user_agent) > -1
```

This will check if there is an exact match with `user_agent` in the `bots_list`. What you'd be looking for is something like 'facebookexternalhit/1.1' or 'Pinterest/0.1 +http://pinterest.com/'.

Tomtomgo commented:
@riyadhalnur thanks a lot! Currently this won't work, but I like this feature, so if you fix it I'll be glad to accept the PR! :+1:

riyadhalnur commented:
@Tomtomgo Thanks. I'll make the fixes.

riyadhalnur commented:
@Tomtomgo Done

Tomtomgo commented on index.js:
```diff
@@ -6,8 +6,10 @@
     function is_robot(user_agent) {
-        // Check for the simple 'bot' string.
-        return user_agent.toLowerCase().indexOf('bot') > -1;
+        var bots_list = [/baiduspider/, /facebookexternalhit/, /embedly/, /quora/, /link/, /preview/, /outbrain/, /pinterest/, /vkShare/, /W3C_Validator/];
```
Regexes should be case-insensitive I think, e.g. Baidu spider & Pinterest have upper case in their user agent strings.

riyadhalnur committed:
Make regex search case-insensitive

Some spiders/bots may have uppercase letters. Make the string lookup case insensitive in that case to get around it.

riyadhalnur commented:
@Tomtomgo Thank you for noticing that. I added the `i` flag to the regexes to make the search case-insensitive.

Tomtomgo commented:
Merged, thanks a lot @riyadhalnur! :tada:

riyadhalnur commented:
@Tomtomgo Can you make a new release for the package with these changes?

Tomtomgo commented:
Definitely, done. New version on https://www.npmjs.com/package/phearjs-express.

riyadhalnur commented:
@Tomtomgo Thank you :smiley:
